### ISSUE TITLE
Query Store stuff

### ISSUE THREAD

**Issue by BlitzErik** [BrentOzarULTD/SQL-Server-First-Responder-Kit#904]
With the initial code for a query store stored proc written, here are some questions I'm still mulling over:

- Should we with with sp_BlitzQS for the name?
- We collect compile, bind, and optimize cpu, duration, and memory. What are thresholds that we should consider bad or high for those?
- What QS info should we give back to users? Size? Age? Metadata? 
- Should we tell me their worst dates for perf for each of the categories we gather interval data for?
- Should we look for actual cached plans, and note if the plan isn't in the plan cache? Is there any plan cache (or views used in BlitzCache) we could hit for more info? (As a for-instance, we don't get the same memory grant information from QS. We get used, but not how much we got, so we can't flag unused memory grants the same way
- Are there any other filtering or mode parameters we want to pass in?
- Should we raise a warning based on compile count? if compile count is higher than execution count?
- How do we want to address the wait stats DMV in 2017?
- Are there any other parameter sniffing characteristics we should be looking for?
- Since we don't have to worry about version-checking, should we investigate selective XML indexes to speed up XML processing?
- Should we standardize all inserted input? Right now, we take data as-is, which is logged in microseconds, and 8k pages. We don't care about most things until they're at the millisecond or some MB threshold, but that's fairly simple to grab on the fly.

**Comment by BrentOzar** [comment]
Thinking out loud about a few of these:

- Should we go with sp_BlitzQS for the name? - I would go with sp_BlitzQueryStore just to make it more obvious what it's about. I know, it's gonna be more typing, but it's not like you'd type this one as often as sp_BlitzWho.
- We collect compile, bind, and optimize cpu, duration, and memory. What are thresholds that we should consider bad or high for those? - I don't think we should flag any of those until we've got a better handle on what's normal. I have no clue either. If you put a gun to my head, I'd say 5 seconds of CPU or duration (because that's a pretty big deal).
- Should we tell me their worst dates for perf for each of the categories we gather interval data for? - Yeah, I absolutely love that.
- Should we look for actual cached plans, and note if the plan isn't in the plan cache? Is there any plan cache (or views used in BlitzCache) we could hit for more info? (As a for-instance, we don't get the same memory grant information from QS. We get used, but not how much we got, so we can't flag unused memory grants the same way - OOO, neat, that'd be awesome! Or just give them the more-info query from sp_BlitzCache so they can go analyze the current plan?
- Should we raise a warning based on compile count? if compile count is higher than execution count? - Even if it's 50% of execution count, I'd start to get worried.
- Since we don't have to worry about version-checking, should we investigate selective XML indexes to speed up XML processing? - I would just make an issue for that for later. It's beyond an MVP. It's a great idea though.
- Should we standardize all inserted input? Right now, we take data as-is, which is logged in microseconds, and 8k pages. We don't care about most things until they're at the millisecond or some MB threshold, but that's fairly simple to grab on the fly. - OOO, yeah, make it easier for folks to interpret. I agree that milliseconds and MB make more sense.

**Comment by BlitzErik** [comment]
Thinking out loud about a few of these:

- [X ] - Should we go with sp_BlitzQS for the name? - I would go with sp_BlitzQueryStore just to make it more obvious what it's about. I know, it's gonna be more typing, but it's not like you'd type this one as often as sp_BlitzWho.
- [ ] - We collect compile, bind, and optimize cpu, duration, and memory. What are thresholds that we should consider bad or high for those? - I don't think we should flag any of those until we've got a better handle on what's normal. I have no clue either. If you put a gun to my head, I'd say 5 seconds of CPU or duration (because that's a pretty big deal).
- [ ] - Should we tell me their worst dates for perf for each of the categories we gather interval data for? - Yeah, I absolutely love that.
- [X ] - Should we look for actual cached plans, and note if the plan isn't in the plan cache? Is there any plan cache (or views used in BlitzCache) we could hit for more info? (As a for-instance, we don't get the same memory grant information from QS. We get used, but not how much we got, so we can't flag unused memory grants the same way - OOO, neat, that'd be awesome! Or just give them the more-info query from sp_BlitzCache so they can go analyze the current plan?
- [ ] - Should we raise a warning based on compile count? if compile count is higher than execution count? - Even if it's 50% of execution count, I'd start to get worried.
- [X ] - Should we standardize all inserted input? Right now, we take data as-is, which is logged in microseconds, and 8k pages. We don't care about most things until they're at the millisecond or some MB threshold, but that's fairly simple to grab on the fly. - OOO, yeah, make it easier for folks to interpret. I agree that milliseconds and MB make more sense.
